src/App.ts

```diff
@@ -969,9 +985,34 @@ export default class App<AppCustomContext extends StringIndexed = StringIndexed>
       }
     }
 
+    /**
+     * Set the Bot Access Token if it exists in event payload to the context.
+     *
+     * A bot_access_token will exist in any payload that has been generated
+     * in the context of a Slack Function execution. This includes function_execution
+     * events themselves, as well as interactivity payloads (e.g. block_actions, views).
+     *
+     * This token must be used for further Slack API calls which are relevant to this
+     * Slack Function execution in order for any interactivity context data to be
+     * properly sent by Slack.
+     *
+     * Bolt will set this value in the event Context and
+     * use this token in place of any other token (for example the token
+     * that the App was configured with) to initialize the Web Client.
+     *
+     * Sometimes the bot_access_token is located in the event
+     * Sometimes it is located directly in the body.
+     */
+    let botAccessToken = body.bot_access_token;
+    if (botAccessToken === undefined && 'event' in body) {
+      botAccessToken = body.event.bot_access_token;
+    }
+
+    // Declare the event context
     const context: Context = {
       ...authorizeResult,
       ...event.customProperties,
+      botAccessToken,
```

Comment:
@seratch Great point on how the name doesn't really describe what it is and potential confusion that may come up. 

I think I have to go with something along the lines of approach 1 to avoid another potential confusion - the fact that `botToken` should represent an `xoxb-` prefaced string when bot_access_token should always be a `xwfp-` prefaced string 😭. 

because I don't want to drop the naming relationship to bot_access_token and because function by itself still seems to general - `slackFunctionBotAccessToken` is the compromise and probably the most correct in terms of "it lets your SlackFunction access a BotToken". 

Thankfully devs shouldn't have to set this token themselves, as we take care of setting it.

Reply:
All make sense here. Thanks for revising this!